airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -577,7 +577,9 @@ def build_pod_request_obj(self, context: Context | None = None) -> k8s.V1Pod:
             pod.metadata.name = PodGenerator.make_unique_pod_id(pod.metadata.name)
 
         if not pod.metadata.namespace:
-            hook_namespace = self.hook.conn_extras.get("extra__kubernetes__namespace")
+            # todo: replace with call to `hook.get_namespace` in 6.0, when it doesn't default to `default`.
+            # if namespace not actually defined in hook, we want to check k8s if in cluster
+            hook_namespace = self.hook._get_namespace()
```

Comment:
IMO, this method should be made *public*?

Reply:
yes it should. it will happen after 6.0 when we replace the current `get_namespace`

Reply:
Should there be some test to validate or we already have one?

Reply:
yes, we have one

Reply:
you mean to remind us to update those in 6.0? yes, we do

Reply:
(and then when user removes _get_namespace, they'll see this reference and presumably read the todo and update it)